torch/csrc/jit/runtime/interpreter.cpp

```diff
@@ -701,6 +701,22 @@ struct CodeImpl {
     return r;
   }
 
+  void emitNumInputsOperand(int numInputs) {
+    int result = constant_table_.size();
+    auto ival = IValue(numInputs);
+    // value_to_reg_[ival] = result;
+    constant_table_.emplace_back(std::move(ival));
+    insertInstruction(LOADC, result);
+  }
+
+  void emitTypeCheck(Node* node) {
+    emitLoadInputs(node->inputs());
+    // Since TypeCheck is variadic. Insert an instruction to push number of
+    // input operands into the stack.
+    emitNumInputsOperand(node->inputs().size());
+    insertInstruction(TYPECHECK, emitType(node->output(0)->type()));
```

Comment:
Do we want to support typecheck with 0 inputs/outputs? If we do (we seem to handle that case in the interpreter code), then we need to check for that case here as well.

Reply:
zero case is degenerate and I think it should not supported. I am adding the an assertion in emitTypeCheck.